airflow/contrib/hooks/opsgenie_alert_hook.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import json
+
+from airflow.hooks.http_hook import HttpHook
+from airflow.exceptions import AirflowException
+
+
+class OpsgenieAlertHook(HttpHook):
+    """
+    This hook allows you to post alerts to Opsgenie.
+    Accepts a connection that has an Opsgenie API key as the connection's password.
+    Each Opsgenie API key can be pre-configured to a team integration.
+    You can override these defaults in this hook.
+
+    :param http_conn_id: Http connection ID with host as "https://api.opsgenie.com/"
+                      and Opsgenie API key as the connection's password
+                      (e.g. "eb243592-faa2-4ba2-a551q-1afdf565c889")
+    :type http_conn_id: str
+    :param payload: Opsgenie API Create Alert payload values
+                    See https://docs.opsgenie.com/docs/alert-api#section-create-alert
+    :type payload: dict
+    :param proxy: Proxy to use to make the Opsgenie Alert API call
+    :type proxy: str
+    """
+    def __init__(self,
+                 http_conn_id=None,
+                 payload={},
+                 proxy=None,
+                 *args,
+                 **kwargs
+                 ):
+        super(OpsgenieAlertHook, self).__init__(*args, **kwargs)
+        self.http_conn_id = http_conn_id
+        self.api_key = None
+        self.payload = payload
+        self.proxy = proxy
+
+    def _get_api_key(self, http_conn_id):
+        """
+        Given a conn_id, return the api_key to use
+        :param http_conn_id: The conn_id provided
+        :type http_conn_id: str
+        :return: api_key (str) to use
+        """
+        if http_conn_id:
+            conn = self.get_connection(http_conn_id)
+            return conn.password
```

Comment:
@zhongjiajie I originally had it as `extra -> api_key`, however when you do so it will log it as clear text in the logs.

However [passwords](https://github.com/apache/airflow/blob/master/airflow/models/connection.py#L266) are obfuscated in the logs.

Reply:
You rigth, we should keep in `password` to not show to logs